DigitalLearningSolutions.Data/Services/CustomPromptsService.cs

```diff
@@ -26,21 +31,138 @@ public CustomPromptsService(ICustomPromptsDataService customPromptsDataService)
 
             var result = customPromptsDataService.GetCentreCustomPromptsByCentreId(centreId.Value);
 
-            return new CentreCustomPrompts
+            return new CentreCustomPrompts(result.CentreId, PopulateCustomPromptListFromCentreCustomPromptsResult(result));
+        }
+
+        public CentreCustomPromptsWithAnswers? GetCentreCustomPromptsWithAnswersByCentreIdAndDelegateUser(int? centreId,
+            DelegateUser? delegateUser)
+        {
+            if (centreId == null || delegateUser == null)
+            {
+                return null;
+            }
+
+            var result = customPromptsDataService.GetCentreCustomPromptsByCentreId(centreId.Value);
+
+            return new CentreCustomPromptsWithAnswers(result.CentreId, PopulateCustomPromptWithAnswerListFromCentreCustomPromptsResult(result, delegateUser));
+        }
+
+        private List<CustomPrompt> PopulateCustomPromptListFromCentreCustomPromptsResult(CentreCustomPromptsResult? result)
+        {
+            var list = new List<CustomPrompt>();
+
+            if (result == null)
+            {
+                return list;
+            }
+
+            var prompt1 = PopulateCustomPrompt(1, result.CustomField1Prompt, result.CustomField1Options,
+                result.CustomField1Mandatory);
+            if (prompt1 != null)
+            {
+                list.Add(prompt1);
+            }
+
+            var prompt2 = PopulateCustomPrompt(2, result.CustomField2Prompt, result.CustomField2Options,
+                result.CustomField2Mandatory);
+            if (prompt2 != null)
+            {
+                list.Add(prompt2);
+            }
+
+            var prompt3 = PopulateCustomPrompt(3, result.CustomField3Prompt, result.CustomField3Options,
+                result.CustomField3Mandatory);
+            if (prompt3 != null)
+            {
+                list.Add(prompt3);
+            }
+
+            var prompt4 = PopulateCustomPrompt(4, result.CustomField4Prompt, result.CustomField4Options,
+                result.CustomField4Mandatory);
+            if (prompt4 != null)
+            {
+                list.Add(prompt4);
+            }
+
+            var prompt5 = PopulateCustomPrompt(5, result.CustomField5Prompt, result.CustomField5Options,
+                result.CustomField5Mandatory);
+            if (prompt5 != null)
+            {
+                list.Add(prompt5);
+            }
+
+            var prompt6 = PopulateCustomPrompt(6, result.CustomField6Prompt, result.CustomField6Options,
+                result.CustomField6Mandatory);
+            if (prompt6 != null)
+            {
+                list.Add(prompt6);
+            }
+
+            return list;
+        }
+
+        private List<CustomPromptWithAnswer> PopulateCustomPromptWithAnswerListFromCentreCustomPromptsResult(CentreCustomPromptsResult? result, DelegateUser delegateUser)
```

Comment:
Ah, shame there isn't a nicer way of doing all this! One minor thing - I think the populate methods can be made static :)